itests/worker_test.go

```diff
@@ -401,3 +404,75 @@ func TestWindowPostWorkerManualPoSt(t *testing.T) {
 	require.NoError(t, err)
 	require.Len(t, lastPending, 0)
 }
+
+func TestSchedulerRemoveRequest(t *testing.T) {
+	ctx := context.Background()
+	_, miner, worker, ens := kit.EnsembleWorker(t, kit.WithAllSubsystems(), kit.ThroughRPC(), kit.WithNoLocalSealing(true),
+		kit.WithTaskTypes([]sealtasks.TaskType{sealtasks.TTFetch, sealtasks.TTCommit1, sealtasks.TTFinalize, sealtasks.TTDataCid, sealtasks.TTAddPiece, sealtasks.TTPreCommit1, sealtasks.TTCommit2, sealtasks.TTUnseal})) // no mock proofs
+
+	ens.InterconnectAll().BeginMining(50 * time.Millisecond)
+
+	e, err := worker.Enabled(ctx)
+	require.NoError(t, err)
+	require.True(t, e)
+
+	type SchedDiagRequestInfo struct {
+		Sector   abi.SectorID
+		TaskType sealtasks.TaskType
+		Priority int
+		SchedId  uuid.UUID
+	}
+	type SchedDiagInfo struct {
+		Requests    []SchedDiagRequestInfo
+		OpenWindows []string
+	}
+	type i struct {
+		SchedInfo    interface{}
+		ReturnedWork []string
+		Waiting      []string
+		CallToWork   map[string]string
+		EarlyRet     []string
+	}
+
+	go miner.PledgeSectors(ctx, 1, 0, nil)
+	// Keep checking till sector state is PC2, the request should get stuck as worker cannot process PC2
+	for {
+		st, err := miner.SectorsStatus(ctx, abi.SectorNumber(1), false)
```

Comment:
You likely want to call miner.SectorListNonGenesis() to get the correct sector number. Sectors 0/1 are going to be presealed genesis sectors

Reply:
@magik6k I actually updated the code a little bit. I am using StartPledge() as it gives back a map of pledged sector and thus no more guessing game. 
I am stuck with casting the interface{} to struct to extract the data of SealingSchedDiag to get UUID. Can you help with that? 
I will add the sleep once I am able to figure out this data extraction.

Reply:
Test fixed